src/Polyfills/AssertContainsOnly.php

```diff
@@ -0,0 +1,472 @@
+<?php
+
+namespace Yoast\PHPUnitPolyfills\Polyfills;
+
+use PHPUnit\SebastianBergmann\Exporter\Exporter as Exporter_In_Phar_Old;
+use PHPUnitPHAR\SebastianBergmann\Exporter\Exporter as Exporter_In_Phar;
+use SebastianBergmann\Exporter\Exporter;
+use Traversable;
+use Yoast\PHPUnitPolyfills\Autoload;
+use Yoast\PHPUnitPolyfills\Helpers\ResourceHelper;
+
+/**
+ * Polyfill the assertContainsNotOnlyInstancesOf(), assertContainsOnlyArray(), assertContainsOnlyBool(),
+ * assertContainsOnlyCallable(), assertContainsOnlyFloat(), assertContainsOnlyInt(),
+ * assertContainsOnlyIterable(), assertContainsOnlyNull(), assertContainsOnlyNumeric(),
+ * assertContainsOnlyObject(), assertContainsOnlyResource(), assertContainsOnlyClosedResource(),
+ * assertContainsOnlyScalar(), assertContainsOnlyString(), assertContainsNotOnlyArray(),
+ * assertContainsNotOnlyBool(), assertContainsNotOnlyCallable(), assertContainsNotOnlyFloat(),
+ * assertContainsNotOnlyInt(), assertContainsNotOnlyIterable(), assertContainsNotOnlyNull(),
+ * assertContainsNotOnlyNumeric(), assertContainsNotOnlyObject(), assertContainsNotOnlyResource(),
+ * assertContainsNotOnlyClosedResource(), assertContainsNotOnlyScalar(), and assertContainsNotOnlyString() methods.
+ *
+ * Introduced in PHPUnit 11.5.0.
+ *
+ * @link https://github.com/sebastianbergmann/phpunit/commit/a726e0396e71cc77bc0b459f93481c29e726dbd8
+ *
+ * @since 3.1.0
+ */
+trait AssertContainsOnly {
+
+	/**
+	 * Asserts that $haystack does not only contain instances of class or interface $type.
+	 *
+	 * @param string          $type     Class or interface name.
+	 * @param iterable<mixed> $haystack The variable to test.
+	 * @param string          $message  Optional failure message to display.
+	 *
+	 * @return void
+	 */
+	final public static function assertContainsNotOnlyInstancesOf( string $type, $haystack, string $message = '' ) {
```

Comment:
Why list `assertContainsNotOnlyInstancesOf()` first? In the `Framework/Assert.php` file, it's listed after `assertContainsNotOnlyString()`. Wondering if placement should match PHPUnit for comparison? What do you think @jrfnl?

Reply:
I haven't used the order in the `Framework/Assert.php` file as a guideline anywhere, though at times the assertion order may align, in which case it's more by accident than by design.

In this case, there are two - very arbitrary - reasons for the order:
1. It was listed first in the [changelog of PHPUnit 11.5.0](https://github.com/sebastianbergmann/phpunit/blob/11.5/ChangeLog-11.5.md#1150---2024-12-06) ;-)
2. All the other assertions are basically sets of two assertions, while this is a single new assertion. Using some form of alphabetic ordering would bury the assertion in the middle of those sets making this "odd one out" hard to find. Which leaves having it either as the first or the last method in the class and I, arbitrarily, choose to place it as the first.

Reply:
Yeah, I too noticed it was listed first in the changelog. That's a good point. 

I followed the PHPUnit file during the review, which caught my attention. Point 2 makes sense.

Thanks for sharing your reasons.